Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -594,7 +597,11 @@ void doDepositionSharedShapeN (const GetParticlePosition<PIdx>& GetPosition,
             const unsigned int ip = permutation[ip_orig];
             depositComponent<depos_order>(GetPosition, wp, uxp, uyp, uzp, ion_lev, jx_buff, jx_type,
                                           relative_time, dinv, xyzmin, lo, q, n_rz_azimuthal_modes,
-                                          ip, zdir, NODE, CELL, 0);
+                                          ip,
+#if defined(WARPX_ZINDEX)
+                                          zdir,
+#endif
+                                          NODE, CELL, 0);
```

Comment:
I still don't love these kinds of `if` conditions in the middle of the list of a function's arguments, but I don't have in mind an alternative right now - and I think we do this in many places in the code.